src/main/java/de/numcodex/feasibility_gui_backend/query/broker/direct/DirectSpringConfig.java

```diff
@@ -9,23 +11,36 @@
 import org.springframework.web.reactive.function.client.WebClient;
 
 /**
- * Spring configuration for providing a {@link DirectBrokerClient} instance.
+ * Spring configuration for providing a {@link DirectBrokerClientCql} instance.
  */
 @Lazy
 @Configuration
 public class DirectSpringConfig {
 
+    @Value("${app.broker.direct.useCql:false}")
+    private boolean useCql;
+
     @Value("${app.flare.baseUrl}")
     private String flareBaseUrl;
 
+    @Value("${app.cql.baseUrl}")
+    private String cqlBaseUrl;
+
     @Qualifier("direct")
     @Bean
     public BrokerClient directBrokerClient(WebClient directWebClient) {
-        return new DirectBrokerClient(directWebClient);
+        if (useCql) {
+            FhirContext fhirContext = FhirContext.forR4();
+            IGenericClient fhirClient = fhirContext.newRestfulGenericClient(cqlBaseUrl);
+            return new DirectBrokerClientCql(fhirContext, fhirClient);
+        } else {
+            return new DirectBrokerClientFlare(directWebClient);
+        }
     }
 
     @Bean
     public WebClient directWebClient() {
```

Comment:
Please add flare here or remove the method.

Reply:
Not sure if I get what you mean. Just changing the name of the Method?